api/v1beta1/specialresource_types.go

```diff
@@ -54,94 +58,127 @@ type SpecialResourceArtifacts struct {
 	Claims []SpecialResourceClaims `json:"claims,omitempty"`
 }
 
-// SpecialResourceBuildArgs defines the observed state of SpecialResource
+// SpecialResourceBuildArgs is not used.
+// TODO(qbarrand) check if this type can be removed
 type SpecialResourceBuildArgs struct {
 	Name  string `json:"name"`
 	Value string `json:"value"`
 }
 
-// SpecialResourceConfiguration defines the observed state of SpecialResource
+// SpecialResourceConfiguration is not used.
+// TODO(qbarrand) check if this type can be removed
 type SpecialResourceConfiguration struct {
 	Name  string   `json:"name"`
 	Value []string `json:"value"`
 }
 
-// SpecialResourceGit defines the observed state of SpecialResource
+// SpecialResourceGit is not used.
+// TODO(qbarrand) check if this type can be removed
 type SpecialResourceGit struct {
 	Ref string `json:"ref"`
 	Uri string `json:"uri"`
 }
 
-// SpecialResourceSource defines the observed state of SpecialResource
+// SpecialResourceSource is not used.
+// TODO(qbarrand) check if this type can be removed
 type SpecialResourceSource struct {
 	Git SpecialResourceGit `json:"git,omitempty"`
 }
 
-// SpecialResourceDriverContainer defines the desired state of SpecialResource
+// SpecialResourceDriverContainer is not used.
+// TODO(qbarrand) check if this type can be removed
 type SpecialResourceDriverContainer struct {
 	// +kubebuilder:validation:Optional
 	Source SpecialResourceSource `json:"source,omitempty"`
+
 	// +kubebuilder:validation:Optional
 	Artifacts SpecialResourceArtifacts `json:"artifacts,omitempty"`
 }
 
-// SpecialResourceSpec defines the desired state of SpecialResource
+// SpecialResourceSpec describes the desired state of the resource, such as the recipe to be used and a selector
+// on which nodes it should be installed.
+// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+// +kubebuilder:validation:Required
 type SpecialResourceSpec struct {
+	// Chart describes the Helm chart that needs to be installed.
```

Comment:
Not sure about the duality of chart vs recipe here and there. Maybe we should use the same word?

Reply:
[OpenShift docs](https://docs.openshift.com/container-platform/4.9/hardware_enablement/psap-special-resource-operator.html) use `chart`, so let us do that here as well.